lte/gateway/c/oai/tasks/mme_app/mme_app_ip_imsi.cpp

```diff
@@ -15,24 +15,26 @@ limitations under the License.
 #include <unordered_map>
 #include <iostream>
 #include "mme_app_ip_imsi.h"
-using namespace std;
-typedef unordered_map<string, vector<uint64_t>> Ipv4Map;
-
-Ipv4Map ipv4map;
+#include "mme_app_ueip_imsi_map.h"
+#include "mme_app_state_manager.h"
 
-void initialize_ipv4_map() {
-  OAILOG_FUNC_IN(LOG_MME_APP);
-  ipv4map = Ipv4Map{};
-  OAILOG_FUNC_OUT(LOG_MME_APP);
-}
+using namespace std;
 
 void mme_app_log_ipv4_imsi_map() {
   OAILOG_FUNC_IN(LOG_MME_APP);
-  for (auto itr = ipv4map.begin(); itr != ipv4map.end(); ++itr) {
-    for (auto it_vec = itr->second.begin(); it_vec != itr->second.end();
+  UeIpImsiMap* ueip_imsi_map =
```

Comment:
use `UeIpImsiMap&` instead of pointer and update the method signature accordingly.